lib/remap-functions/src/parse_key_value.rs

```diff
@@ -0,0 +1,351 @@
+use nom::{
+    self,
+    branch::alt,
+    bytes::complete::{escaped, tag, take_until, take_while1},
+    character::complete::{char, satisfy, space0},
+    combinator::{map, rest},
+    multi::{many1, separated_list1},
+    sequence::{delimited, preceded, tuple},
+    IResult,
+};
+use remap::prelude::*;
+use std::collections::BTreeMap;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseKeyValue;
+
+impl Function for ParseKeyValue {
+    fn identifier(&self) -> &'static str {
+        "parse_key_value"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "field_split",
```

Comment:
What do you think about `field_delimiter` and `key_value_delimiter` instead?

Reply:
Yeah, I think that's better.